### PR TITLE
Add all properties when creating a subscription

### LINKED ISSUE #24469: CalDAVBackend::createSubscription ignores a lot of properties
https://github.com/owncloud/core/blob/master/apps/dav/lib/caldav/caldavbackend.php#L1063-1070

This code only inserts the hardcoded values. Properties like displayname, color, order, stripAlerts and stripAttachments are simply ignored without further notice.

createCalendar inserts all supported properties: https://github.com/owncloud/core/blob/master/apps/dav/lib/caldav/caldavbackend.php#L378-380

### PR DESCRIPTION
**Pull request by tcitworld**
Originally at #25231

Fix #24469

### CODE REVIEW

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 1]
```diff
@@ -1079,22 +1079,27 @@ function createSubscription($principalUri, $uri, array $properties) {
 			'lastmodified' => time(),
 		];
 
-		foreach($this->subscriptionPropertyMap as $xmlName=>$dbName) {
-			if (isset($properties[$xmlName])) {
+		$propertiesBoolean = array('striptodos', 'stripalarms', 'stripattachments');
```
Comment (DeepDiver1975) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: please use short array declaration []

**Comment by DeepDiver1975** [comment]
@georgehrke please test again - thx

File: apps/dav/lib/CalDAV/CalDavBackend.php (hunk 1 above)
Comment (tcitworld) [inline apps/dav/lib/CalDAV/CalDavBackend.php]: Fixed and squashed commits.

**Comment by georgehrke** [comment]
looks good to me 👍

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**